libraries/chain/transaction_context.cpp

```diff
@@ -323,10 +341,13 @@ namespace eosio { namespace chain {
          billing_timer_exception_code = tx_cpu_usage_exceeded::code_value;
       }
 
-      net_usage = ((net_usage + 7)/8)*8; // Round up to nearest multiple of word size (8 bytes)
-
       eager_net_limit = net_limit;
-      check_net_usage();
+
+      if( explicit_net_usage ) {
+         check_net_usage();    // Should already be rounded. Just check NET usage satisfies limits.
+      } else {
+         round_up_net_usage(); // Round up to nearest multiple of word size (8 bytes) and check NET usage satisfies limits.
+      }
 
```

Comment:
I think I would prefer:
```
  round_up_net_usage();
  check_net_usage(); // check NET usage even for explicit_net_usage to verify satisfies limits.
```
To me that is less confusing than trying to understand why this `if` check is here.